### PR TITLE
Update SubmitRequest from sharedb to support CreateOP, DeleteOp and EditOp

### PR DESCRIPTION
**Pull request by deongroenewald**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [sharedb Op Types](https://github.com/share/sharedb/blob/c8275fb49b76c83dead36e2cee305fd349d40d8e/lib/agent.js#L843-L869)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/sharedb/lib/sharedb.d.ts [hunk 1]
```diff
@@ -46,11 +46,15 @@ export interface RawOp {
     src: string;
     seq: number;
     v: number;
-    op: Op[];
     m: any;
     c: string;
     d: string;
 }
+
+export type CreateOp = RawOp & { create: {type: string; data: any }; del: undefined; op: undefined; };
```
Comment (ericyhwang) [inline types/sharedb/lib/sharedb.d.ts:54]: Add a space before `type` - should have been caught by the linter, I'm unsure why it wasn't.
```suggestion
export type CreateOp = RawOp & { create: { type: string; data: any }; del: undefined; op: undefined; };
```

**Review by ericyhwang — CHANGES_REQUESTED** [review]
Thanks! The actual change looks good, I have a couple small testing-related requests.

1. Can you add a few statements in this test case to exercise the RawOp variations? DefinitelyTyped doesn't actually run the "tests", it just checks to make sure that the tests pass type-checking, which is why the tests mostly test accessing properties.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/cef64f0170d7b74f083ae987d27aca7aa200c6ac/types/sharedb/sharedb-tests.ts#L103-L113

2. While you're at it, remove the reference to qunit, whose types aren't actually used:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/cef64f0170d7b74f083ae987d27aca7aa200c6ac/types/sharedb/sharedb-tests.ts#L1

**Review by ericyhwang — APPROVED** [review]
Looks good, thanks for the type updates and the additional tests!

**Comment by deongroenewald** [comment]
Ready to merge

**Pull request merged**